pymc/logprob/binary.py

```diff
@@ -136,3 +136,57 @@ def comparison_logprob(op, values, base_rv, operand, **kwargs):
         logcdf.name = f"{base_rv_op}_logcdf"
 
     return logprob
+
+
+class MeasurableBitwise(MeasurableElemwise):
+    """A placeholder used to specify a log-likelihood for a bitwise operation RV sub-graph."""
+
+    valid_scalar_types = (Invert,)
+
+
+@node_rewriter(tracks=[invert])
+def find_measurable_bitwise(fgraph: FunctionGraph, node: Node) -> Optional[List[MeasurableBitwise]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableBitwise):
+        return None  # pragma: no cover
+
+    base_var = node.inputs[0]
+    if not (
+        base_var.owner
+        and isinstance(base_var.owner.op, MeasurableVariable)
+        and base_var not in rv_map_feature.rv_values
+    ):
+        return None
+
+    if not base_var.dtype.startswith("bool"):
+        raise TypeError("Base RV should be of type boolean")
```

Comment:
```suggestion
        raise TypeError("Base RV should be of type boolean")
```
We shouldn't raise errors, because another rewrite (including a user custom one) could implement a transformation for this type of variable.

One thing we might want to start doing sooner or latter is to add `log.debug` statements so that users can figure out why didn't pymc figure out the logprob for their expression (where here we would say that `bitwise` for integer variables is not supported by this rewrite).

Reply:
Opened an issue https://github.com/pymc-devs/pymc/issues/6691

Here we should just return `None`, which means the current rewrite does not apply

Reply:
> We shouldn't raise errors, because another rewrite (including a user custom one) could implement a transformation for this type of variable.

Understood. I'll replace it to return None.

> One thing we might want to start doing sooner or later is to add log.debug statements so that users can figure out why didn't pymc figure out the logprob for their expression (where here we could say that bitwise for integer variables is not supported by this rewrite).

I have a question about this. Since the logprob rewrites are in the IR graph, should the user be exposed to the internal errors? (e.g.: If the value variable is potentially measurable).

Reply:
> Since the logprob rewrites are in the IR graph, should the user be exposed to the internal errors? (e.g.: If the value variable is potentially measurable).

What do you mean by internal errors?

Reply:
I was thinking of a scenario when we check the potential measurability of the value variables and whether the user would be concerned with the error. But I figured that it has to be the expression defined by the user that would cause the error, so the answer should be yes. It may have been an oversight on my part since I'm still getting acquainted with the use cases of PyMC.

Reply:
The point I was trying to make is that these rewrites are local. So what looks like something invalid to them, may be valid for another rewrite (ours or provided by the user). So we shouldn't raise any errors ourselves, but simply return `None`, which states: this rewrite does not apply to this graph.

What I suggested in the linking issue, is that we can issue a log message saying why the rewrite was not applied, as it can be very useful for the user to understand why the logprob was not derived for their graph.

Reply:
Okay, I understand now. Thank you so much for clarifying!